proof/crefine/ARM_HYP/Finalise_C.thy

```diff
@@ -2179,29 +2179,38 @@ lemma associateVCPUTCB_ccorres:
           apply ceqv
          apply (rule ccorres_move_c_guard_vcpu)
          apply clarsimp
-         apply (subst ccorres_seq_skip'[symmetric])
          apply (rule ccorres_split_nothrow[where r'=dc and xf'=xfdc])
-             apply (rule  setObject_vcpuTCB_updated_Basic_ccorres[where tptr="Some tptr" and t="Map.empty"
-                                                                 , simplified option_to_ctcb_ptr_def, simplified])
+             apply (rule setObject_vcpuTCB_updated_Basic_ccorres[where tptr="Some tptr" and t="Map.empty",
+                                                                 simplified option_to_ctcb_ptr_def, simplified])
             apply ceqv
-           apply (rule ccorres_return_Skip)
-          apply (rule wp_post_taut)
+           apply (rule ccorres_pre_getCurThread, rename_tac curThread)
+           apply (subst ccorres_seq_skip'[symmetric])
+           apply (rule ccorres_split_nothrow[where r'=dc and xf'=xfdc])
+               apply (rule_tac R="\<lambda>s. curThread = ksCurThread s" in ccorres_when)
+                apply (intro allI impI)
+                apply (unfold mem_simps)[1]
+                apply (clarsimp simp: rf_sr_ksCurThread)
```

Comment:
This is a bit odd, looks like a fight with clarsimp, and I'm not sure what `mem_simps` is exactly, but  it's an awkward-looking combination.

Reply:
Do those three applies then collapse into one that solves the subgoal?

Reply:
Yes!